satpy/tests/reader_tests/test_eps_l1b.py

```diff
@@ -160,6 +163,63 @@ def mock_getitem(key):
             sun_zen_np2 = np.array(avhrr_reader.sun_zen)
             assert np.allclose(sun_zen_np1, sun_zen_np2)
 
+
+class TestWrongEPSL1B(TestCase):
+    """Test the filehandler on a corrupt file."""
+
+    @pytest.fixture(autouse=True)
+    def inject_fixtures(self, caplog):
+        """Inject caplog."""
+        self._caplog = caplog
+
+    def setUp(self):
+        """Set up the tests."""
+        # ipr is not present in the xml format ?
+        structure = [(1, ('mphr', 0)), (1, ('sphr', 0)), (11, ('ipr', 0)),
+                     (1, ('geadr', 1)), (1, ('geadr', 2)), (1, ('geadr', 3)),
+                     (1, ('geadr', 4)), (1, ('geadr', 5)), (1, ('geadr', 6)),
+                     (1, ('geadr', 7)), (1, ('giadr', 1)), (1, ('giadr', 2)),
+                     (1, ('veadr', 1)), (1080, ('mdr', 2))]
+
+        sections = create_sections(structure)
+        # Wrong number of lines, should be 1080 :)
+        sections[('mphr', 0)]['TOTAL_MDR'] = b'TOTAL_MDR                     =   1078\n'
+        sections[('mphr', 0)]['SPACECRAFT_ID'] = b'SPACECRAFT_ID                 = M03\n'
+        sections[('mphr', 0)]['INSTRUMENT_ID'] = b'INSTRUMENT_ID                 = AVHR\n'
+        sections[('sphr', 0)]['EARTH_VIEWS_PER_SCANLINE'] = b'EARTH_VIEWS_PER_SCANLINE      =  2048\n'
```

Comment:
Same here, use `self.earth_views_per_scanline` and string formatting here.

Reply:
Yep, nice!

Reply:
done